code/modules/mob/living/carbon/xenomorph/abilities/queen/queen_powers.dm

```diff
@@ -437,35 +440,66 @@
 	var/mob/living/carbon/xenomorph/queen/user_xeno = owner
 	if(user_xeno.hive.hivenumber != XENO_HIVE_CORRUPTED)
 		return
+
 	if(!user_xeno.check_state())
 		return
+
 	var/datum/hive_status/corrupted/hive = user_xeno.hive
-	hive.sanitize_allies()
+
 	if(!length(hive.personal_allies))
 		to_chat(user_xeno, SPAN_WARNING("We don't have personal allies."))
 		return
-	var/mob/living/carbon/target_mob = tgui_input_list(usr, "Target", "Break the Personal Alliance With...", hive.personal_allies, theme="hive_status")
-	if(!target_mob) return
+
+	var/list/mob/living/allies = list()
+	var/list/datum/weakref/dead_refs = list()
+	for(var/datum/weakref/ally_ref in hive.personal_allies)
+		var/mob/living/ally = ally_ref.resolve()
+		if(istype(ally))
```

Comment:
```suggestion
		if(ally)
```
`/datum/weakref/proc/resolve` both checks if its qdeleted (which involves a null check) and if the weak_reference circular reference still points back to this weakref, so type checking it should not be necessary. But do try it out.

Reply:
Well, it's working without issues. Tested this proc, deleted the mob and it didn't appear.